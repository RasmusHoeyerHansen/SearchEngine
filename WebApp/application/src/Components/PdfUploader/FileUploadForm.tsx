import {ChangeEvent} from "react";

export interface IFileUploadFormProps extends IFileUploadFormOptionalProps {
    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

interface IFileUploadFormOptionalProps {
    fileTypes?: [ValidFileTypes]
}

type ValidFileTypes = 'application/pdf' | 'application/vnd.ms-excel';

export const FileUploadForm = ({
                                   onChange, fileTypes = ['application/pdf']
                               }: IFileUploadFormProps) => {
    let acceptFileTypes: string = fileTypes.join(',');
    return (<form id="FileUploadForm">
        <input id={'FileUploadForm-input'}
               type="file"
               name="formFile" multiple
               accept={acceptFileTypes}
               onChange={onChange}
        />
        <input id={"FileUploadForm-submit"} type={"submit"}/>
        <label htmlFor="pdfInputForm"/>
    </form>);
}
