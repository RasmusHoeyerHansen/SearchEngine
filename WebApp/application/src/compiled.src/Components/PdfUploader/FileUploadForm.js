var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx } from "react/jsx-runtime";
import { FileInput } from "./FileInput";
export var FileUploadForm = function (_a) {
    var onChange = _a.onChange, _b = _a.fileTypes, fileTypes = _b === void 0 ? ['application/pdf'] : _b;
    var acceptFileTypes = fileTypes.join(',');
    return _jsx("form", __assign({ id: "FileUploadForm", role: 'form', "data-testid": 'FileUploadForm' }, { children: _jsx(FileInput, { acceptedFileString: acceptFileTypes, onChange: onChange }, void 0) }), void 0);
};
