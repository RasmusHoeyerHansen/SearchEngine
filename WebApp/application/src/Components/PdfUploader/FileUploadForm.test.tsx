import React from 'react'
import {fireEvent, getByRole, render} from '@testing-library/react'
import {FileUploadForm} from "./FileUploadForm";


describe("FileUploadForm", () => {


    beforeEach(() => {
        jest.clearAllMocks();
    });
    it("renders correctly with default pdf as filetype and matches" +
        " snapshot", () => {
        let {container} = render(<FileUploadForm onChange={() => {
            return;
        }}/>)
        expect(container).toMatchSnapshot();
    });


    it('should render a form element', () => {
        let {container} = render(<FileUploadForm onChange={() => {
            return;
        }}/>);

        // eslint-disable-next-line testing-library/no-node-access,testing-library/no-container
        const element = container.querySelector('#FileUploadForm');
        expect(element).toBeInstanceOf(HTMLFormElement)
    });
    it('should render an file input element', () => {
        let {container} = render(<FileUploadForm onChange={() => {
            return;
        }}/>);

        // eslint-disable-next-line testing-library/no-node-access,testing-library/no-container
        const inputElement = container.querySelector('#FileUploadForm-input') as HTMLInputElement;
        expect(inputElement).toBeInstanceOf(HTMLInputElement)
        expect(inputElement.type).toBe('file');
        expect(inputElement.multiple).toBe(true);
    });

    it('should be able to render a pdf file', function () {
        let {container} = render(<FileUploadForm onChange={() => {
            return;
        }}/>);

        const file = new File([new ArrayBuffer(1)], 'testfile.pdf');

        // eslint-disable-next-line testing-library/prefer-screen-queries,testing-library/no-node-access,testing-library/no-container
        let fileInput =  container.querySelector("#FileUploadForm-input") as HTMLInputElement;
        fireEvent.change(fileInput, {
            target: {files: [file]}
        });
        expect(fileInput.files).toBeTruthy();
        expect(fileInput.files?.length).toBeTruthy();
        expect(fileInput.files?.length).toBe(1);
    });
    it('should be able to render multiple pdf files', function () {
        let {container} = render(<FileUploadForm onChange={() => {
            return;
        }}/>);
        const file = new File([new ArrayBuffer(1)], 'testfile.pdf');
        const file2 = new File([new ArrayBuffer(1)], 'testfile2.pdf');
        const file3 = new File([new ArrayBuffer(1)], 'testfile3.pdf');

        // get the upload button
        // eslint-disable-next-line testing-library/prefer-screen-queries,testing-library/no-node-access,testing-library/no-container
        let fileInput =  container.querySelector("#FileUploadForm-input") as HTMLInputElement;
        fireEvent.change(fileInput, {
            target: {files: [file, file2, file3]}
        });

        expect(fileInput.files).toBeTruthy();
        expect(fileInput.files?.length).toBeTruthy();
        expect(fileInput.files?.length).toBe(3);
    });

})
