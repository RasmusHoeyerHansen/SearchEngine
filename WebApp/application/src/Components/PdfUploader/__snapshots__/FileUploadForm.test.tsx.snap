// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploadForm renders correctly with default pdf as filetype and matches snapshot 1`] = `
<div>
  <form
    id="FileUploadForm"
  >
    <input
      accept="application/pdf"
      id="FileUploadForm-input"
      multiple=""
      name="formFile"
      type="file"
    />
    <input
      id="FileUploadForm-submit"
      type="submit"
    />
    <label
      for="pdfInputForm"
    />
  </form>
</div>
`;
